### DISCUSSION TITLE
Mark read/write collection as 'new' (Csla 4.xx)

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3258] [Questions] [answered]
We are fetching a collection of items from a legacy database and want to 'insert' the same items collection in a new database.
Could we use DataPortal_Create in a collection object and subsequently call 'base.DataPortal_Create'?

Kind regards

**Comment by rockfordlhotka** [comment, marked as answer]
So you are thinking that the create operation would load the data from the old database, and the insert/update/delete operations would interact with the new database?

That should work, as long as nobody tries to reuse that collection later thinking that its create operation will act normally.

**Comment by kcabral817** [comment]
I am curious about how the connection string would work in this use case?  You would be switching database connections and then contexts (if EF Core) which would require two connection strings and then some type of logic to switch databases in midstream.  I am going to be using multi-tenant configuration for my application, and found an article you might be interested in.  https://learn.microsoft.com/en-us/ef/core/miscellaneous/multitenancy

**Reply by Chicagoan2016** [reply]
@kcabral817  This is Csla 4.xx project, we are using ContextManager classes. It's relatively simple because we are hitting the lagacy database  as 'read-only'. All inserts and updates are going to the new (and flashy :)) database.

On a side note, I see 817 in your username, are you from Arlington, TX?

regards

**Comment by kcabral817** [comment]
Lol flashy, that's awesome.  The 817 is for a street number, not an area code.
